src/player_hardcoded_effects.cpp

```diff
@@ -273,6 +273,137 @@ static void eff_fun_hallu( player &u, effect &it )
     }
 }
 
+struct temperature_effect {
+    int str_pen;
+    int dex_pen;
+    int int_pen;
+    int per_pen;
+    // Not translated (static string should not be translated)
+    std::string msg;
+    int msg_chance;
+    // Note: NOT std::string because the pointer is stored so c_str() is not OK
+    // Also not translated
+    const char *miss_msg;
+
+    temperature_effect( int sp, int dp, int ip, int pp, const std::string &ms, int mc,
+                        const char *mm ) :
+        str_pen( sp ), dex_pen( dp ), int_pen( ip ), per_pen( pp ), msg( ms ),
+        msg_chance( mc ), miss_msg( mm ) {
+    }
+
+    void apply( player &u ) const {
+        if( str_pen > 0 ) {
+            u.mod_str_bonus( -str_pen );
+        }
+        if( dex_pen > 0 ) {
+            u.mod_dex_bonus( -dex_pen );
+            u.add_miss_reason( _( miss_msg ), dex_pen );
+        }
+        if( int_pen > 0 ) {
+            u.mod_int_bonus( -int_pen );
+        }
+        if( per_pen > 0 ) {
+            u.mod_per_bonus( -per_pen );
+        }
+        if( !msg.empty() && !u.has_effect( effect_sleep ) && one_in( msg_chance ) ) {
+            u.add_msg_if_player( m_warning, "%s", _( msg.c_str() ) );
+        }
+    }
+};
+
+static void eff_fun_cold( player &u, effect &it )
+{
+    // { body_part, intensity }, { str_pen, dex_pen, int_pen, per_pen, msg, msg_chance, miss_msg }
+    const std::map<std::pair<body_part, int>, temperature_effect> effs = {{
```

Comment:
This probably doesn't matter much since compilers would likely optimize the assignment anyways, but this is a good candidate for being `static`.